### PR TITLE
feat: prepHumanModelForftINIT with paths

### PR DESCRIPTION
**Pull request by johan-gson**
This PR fixes issue #401 and now allows for specifying the paths to the tasks and reactions.tsv. This is good when using the code for for example Mouse-GEM.

### Main improvements in this PR:
prepHumanModelForftINIT supports other models such as Mouse-GEM - the paths to relevant files could not be specified earlier.

**I hereby confirm that I have:**

- [X] Tested my code on my own computer for running the model
- [X] Selected `develop` as a target branch

### CODE REVIEW

**Comment by johan-gson** [comment]
@haowang-bioinfo @mihai-sysbio Could someone of you review this so I can merge it? It is a small change that makes it possible to run ftINIT properly for Mouse-GEM and similar GEMs.

File: code/tINIT/prepHumanModelForftINIT.m
Comment (mihai-sysbio) [inline code/tINIT/prepHumanModelForftINIT.m:1]: With the more generalized scope, I would forego the `Human` word in the function name.

```suggestion
function prepDataHumanGEM = prepHumanModelForftINIT(model, convertGenes, essentialTasksFilePath, rxnsFilePath)
```

File: code/tINIT/prepHumanModelForftINIT.m
Comment (mihai-sysbio) [inline code/tINIT/prepHumanModelForftINIT.m:11]:
```suggestion
%   essentialTasksFilePath  Path to the metabolicTasks_Essential.txt
```

File: code/tINIT/prepHumanModelForftINIT.m
Comment (mihai-sysbio) [inline code/tINIT/prepHumanModelForftINIT.m:14]:
```suggestion
%   rxnsFilePath            Path to the reactions.tsv
```

File: code/tINIT/prepHumanModelForftINIT.m
Comment (mihai-sysbio) [inline code/tINIT/prepHumanModelForftINIT.m:28]: These lines of code would not be needed if no defaults are maintained:
```suggestion
```

**Review by mihai-sysbio — COMMENTED** [review]
Very nice improvement with explicit paths!

In this case,, I find providing default values for the function more a maintenance cost than a benefit for the user. These functions are not meant to be typed that many times in an analysis, so being explicit about what files are used is a small overhead for a user but less code for maintainers to worry about. For example, if the name or path of the metabolic tasks file changes, as a maintainer, one has to remember to find (at least) this occurrence and update it as well to keep the code functional. I've made some inline suggestions accordingly.

**Comment by johan-gson** [comment]
Hi Mihail Although you have a point, I request that we keep it as it is :) The main reason is that we otherwise break the interface, which for example will stop the code we just submitted for the single-cell modeling paper from working. I also have the argument that I don't think all users will know what a task list is, where to find it etc., and that having to supply these params makes using the function unnecessarily difficult - the first time I used tINIT I didn't know what the task file was and where to find it. The whole code around ftINIT in Human-GEM has that only purpose, to make it easier to use ftINIT - the code is not really needed in that sense, it is just a fairly simple wrapper that removes some normally unwanted reactions and fills in some standard values, such as the path to the task file. The reason for this fix is that I didn't think of the case with the mouse model, and that is why these params are needed.

**Comment by mihai-sysbio** [comment]
> The main reason is that we otherwise break the interface, which for example will stop the code we just submitted for the single-cell modeling paper from working.

This PR is to the develop branch, which I hope was not referenced in the submission. So it would only become problematic if it gets to main. Another option would be to keep this PR on hold for a few months.

> I also have the argument that I don't think all users will know what a task list is, where to find it etc., and that having to supply these params makes using the function unnecessarily difficult - the first time I used tINIT I didn't know what the task file was and where to find it. 

This is an important concern. My approach would be to solve this with improved documentation, by continuing to maintain the Human-GEM-guide.

> The whole code around ftINIT in Human-GEM has that only purpose, to make it easier to use ftINIT - the code is not really needed in that sense, it is just a fairly simple wrapper that removes some normally unwanted reactions and fills in some standard values, such as the path to the task file.

We are on the same page here. My approach is not to simplify things towards a magical one command that does everything, but to be very clear about what each command will do.

> The reason for this fix is that I didn't think of the case with the mouse model, and that is why these params are needed.

Now that this branch exists, it can be used even without merging.

**Comment by johan-gson** [comment]
I think you have a point, when I think of it the main concern would be that there is a substantial risk that people now will use the mouse model with another reactions.tsv.

**Comment by mihai-sysbio** [comment]
For the Mouse model, when using the Mouse-GEM reactions.tsv, should people be running `prepHumanModelForftINIT` on Human-GEM or on Mouse-GEM?

And should consider adding an ftINIT page for how to use it with other models, say Mouse-GEM?

**Comment by johan-gson** [comment]
So, they should use it with Mouse-GEM, and with that reactions.tsv, although it probably doesn't matter. But using the right model is important.

File: code/tINIT/prepHumanModelForftINIT.m
Comment (mihai-sysbio) [reply code/tINIT/prepHumanModelForftINIT.m:1]: I am also fine with leaving this as they are now, and postpone any renaming here.

**Review by mihai-sysbio — APPROVED** [review]
@johan-gson please merge when you think is the right time. And a reminder - the `Human-GEM-guide` should probably be updated when this makes it in a release.

**Pull request merged**
